plugins/copr.py

```diff
@@ -473,34 +477,36 @@ def _download_repo(self, project_name, repo_filename):
         api_path = "/coprs/{0}/repo/{1}/dnf.repo?arch={2}".format(project_name, short_chroot, arch)
 
         try:
-            f = self.base.urlopen(self.copr_url + api_path, mode='w+')
-        except IOError as e:
+            response = urlopen(self.copr_url + api_path)
             if os.path.exists(repo_filename):
                 os.remove(repo_filename)
-            if '404' in str(e):
-                if PY3:
-                    import urllib.request
-                    try:
-                        res = urllib.request.urlopen(self.copr_url + "/coprs/" + project_name)
-                        status_code = res.getcode()
-                    except urllib.error.HTTPError as e:
-                        status_code = e.getcode()
-                else:
-                    import urllib
-                    res = urllib.urlopen(self.copr_url + "/coprs/" + project_name)
-                    status_code = res.getcode()
-                if str(status_code) != '404':
-                    raise dnf.exceptions.Error(_("This repository does not have"
-                                                 " any builds yet so you cannot enable it now."))
-                else:
-                    raise dnf.exceptions.Error(_("Such repository does not exist."))
-            raise
-
-        for line in f:
-            if re.match(r"\[copr:", line):
-                repo_filename = os.path.join(self.base.conf.get_reposdir,
-                                             "_" + line[1:-2] + ".repo")
-            break
+        except HTTPError as e:
+            error_msg = _("It wasn't possible to enable this project.\n")
+            error_data = e.headers.get("Copr-Error-Data")
+            error_msg += _("Repository '{0}' does not exist in project '{1}'.").format(
+                '-'.join(self.chroot_parts), project_name)
+            if error_data:
+                error_data_decoded = base64.b64decode(error_data).decode('utf-8')
+                error_data_decoded = json.loads(error_data_decoded)
+                if error_data_decoded.get("available chroots"):
+                    error_msg += _("\nAvailable repositories: ") + ', '.join(
+                        "'{}'".format(x) for x in error_data_decoded["available chroots"])
+                error_msg += _("\n\nIf you want to enable a non-default repository, use the following command:\n"
+                               "  'dnf copr enable {0} <repository>'\n"
+                               "But note that the installed repo file will likely need a manual modification.").format(
+                    project_name)
+                raise dnf.exceptions.Error(error_msg)
+            else:
+                raise dnf.exceptions.Error(error_msg)
```

Comment:
Ok, I observed a weird behavior now .... even though it is probably not a new thing:
```
$ sudo dnf copr enable non/existing  -y
...
Please do not file bug reports about these packages in Fedora
Bugzilla. In case of problems, contact the owner of this repository.
Error: It wasn't possible to enable this project.
Repository 'fedora-35-x86_64' does not exist in project 'non/existing'.
```

The sentence `Repository 'fedora-35-x86_64' does not exist in project 'non/existing'.` could make the user think that the project `non/existing` actually exists, but doesn't have the fedora-35 chroot enabled.

Would you mind special case this, and (when the error_data is None) print something like
`Project 'non/existing' doesn't exist` instead?

Reply:
Agreed. Ideally, instead of:
```
Error: It wasn't possible to enable this project.
Repository 'fedora-35-x86_64' does not exist in project 'non/existing'.
```
You would print:
```
Error: Project 'non/existing' doesn't exist.
```